### PR TITLE
[bitnami/redis] Added support for hostname announcement in cluster setup 

### LINKED ISSUE #23244: [bitnami/redis-cluster] Allow redis cluster to accept connection with hostname
### Name and Version

bitnami/redis-cluster

### What is the problem this feature will solve?

This will allows redis cluster to accept connection with hostname.

### What is the feature you are proposing to solve the problem?

Add Support for 
```
cluster-announce-hostname
cluster-preferred-endpoint-type
```

This will allows redis to run on hostname and fix the tls problem explain here https://github.com/bitnami/charts/issues/14673

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by deveshk0**
### Description of the change

Adds following environment variable to enable support for hostname announcement in cluster setup.

```
REDIS_CLUSTER_ANNOUNCE_HOSTNAME
REDIS_CLUSTER_PREFERRED_ENDPOINT_TYPE
```

### Benefits

- Allows cluster announcement on hostname instead of Pod IP.
- Enables smooth tls enabled cluster

### Possible drawbacks

none

### Applicable issues

- fixes #23244 

### Additional information

None

### CODE REVIEW

**Comment by deveshk0** [comment]
@carrodher @gongomgra any update here?

**Comment by gongomgra** [comment]
HI @deveshk0,

Thanks for your contribution, and sorry for the delay. We are reviewing your changes and will get back to you as soon as possible.

File: bitnami/redis-cluster/README.md [hunk 1]
```diff
@@ -182,6 +182,8 @@ The following env vars are supported for this container:
 | `REDIS_NODES`                           | String delimited by spaces containing the hostnames of all of the nodes that will be part of the cluster                                                           |
 | `REDIS_CLUSTER_ANNOUNCE_IP`             | IP that the node should announce, used for non dynamic ip environents                                                                                              |
 | `REDIS_CLUSTER_ANNOUNCE_PORT`           | Port that the node should announce, used for non dynamic ip environents.                                                                                           |
+| `REDIS_CLUSTER_ANNOUNCE_HOSTNAME`       | Hostname that node should announce, used for non dynamic ip environents.                                                                                           |
```
Comment (gongomgra) [inline bitnami/redis-cluster/README.md:185]: There's a minor typo here `environents` -> `environments`.

**Review by gongomgra — CHANGES_REQUESTED** [review]

File: bitnami/redis-cluster/README.md (hunk 1 above)
Comment (deveshk0) [reply bitnami/redis-cluster/README.md]: Thanks @gongomgra  Updated

**Review by gongomgra — APPROVED** [review]
LGTM!

**Comment by gongomgra** [comment]
Thanks @deveshk0 for your contribution. I'm merging your changes now.

**Pull request merged**